src/snowflake/connector/azure_storage_client.py

```diff
@@ -0,0 +1,255 @@
+#
+# Copyright (c) 2012-2021 Snowflake Computing Inc. All right reserved.
+#
+
+from __future__ import division
+
+import json
+import os
+import xml.etree.cElementTree as ET
+from collections import namedtuple
+from datetime import datetime
+from logging import getLogger
+from random import choice
+from string import hexdigits
+from typing import TYPE_CHECKING, Any, Dict, Optional, Union
+
+from .compat import quote
+from .constants import FileHeader, ResultStatus
+from .encryption_util import EncryptionMetadata
+from .storage_client import SnowflakeStorageClient
+from .vendored import requests
+
+if TYPE_CHECKING:  # pragma: no cover
+    from .file_transfer_agent import SnowflakeFileMeta
+
+logger = getLogger(__name__)
+
+"""
+Azure Location: Azure container name + path
+"""
+AzureLocation = namedtuple(
+    "AzureLocation",
+    ["container_name", "path"],  # Azure container name  # Azure path name
+)
```

Comment:
nit: Use the same class inheritance structure for namedtuples that you used in constants.py